vtex/node-vtex-api
# Change LRUDiskCache to allow subclasses to override functionality

#### What is the purpose of this pull request?

This changes LRUDiskCache so that a subclass can properly override some of its functionality (such as the has/get/set methods, stats calculations, etc).

#### What problem is this solving?

We need to build a new disk cache implementation with extra features. To avoid reimplementing everything again, we can instead extend from LRUDiskCache and override some of its methods to achieve what we need. But the way LRUDiskCache was created didn't allow for these extensions, so this PR is changing it to allow better usage by subclasses.

#### How should this be manually tested?
- Build the codebase and verify that there were no errors.
- Create a new class extending from LRUDiskCache and make sure that there are no errors when overriding the methods.

#### Types of changes
